ls1intum/Artemis
# `Programming exercises`: Fix an issue with user independent repository urls in double blind assessments

Several improvements for participations

1. Significantly reduce response size for participation component (fixes #4488)
2. Create the user independent repository URL for programming participations server side (fixes #4489)
3. Improve code Quality in ParticipationResource
4. Remove lots of client side boilerplate code
5. Convert some client tests from sinon/chai to jest

### Checklist
#### General


#### Server
#### Client
#### Changes affecting Programming Exercises

### Motivation and Context



### Description


### Steps for Testing

Prerequisites:
- 1 Instructor
- 3 Students
- 1 Tutor
- 1 Programming Exercise with manual assessment enabled

1. Instructor: Create a programming exercise
2. Students: Participate in the programming exercise (i.e. start the exercise)
3. Instructor: Navigate to the participations page and check everything is working correctly:
     a. Check that the json call does not include exercise or course objects
     b. Check that the URL behind `Repository Link` is correct (i.e. without username)
4. Tutor: Navigate into the manual assessment of the programming exercise and check that the url behind `Go to repository` does not include a username
5. Instructor: Cleanup the participations
6. Student: Resume the exercise

### Review Progress





#### Code Review
#### Manual Tests

### Test Coverage
no real changes (might slightly increase because unused code was deleted)
